src/beast/beast/nudb/recover.h

```diff
@@ -39,19 +37,22 @@ namespace nudb {
     any partially committed data.
 */
 template <
-    class Hasher = default_hash,
-    class File = native_file>
+    class Hasher,
+    class Codec,
+    class File = native_file,
+    class... Args>
 bool
 recover (
     path_type const& dat_path,
     path_type const& key_path,
     path_type const& log_path,
-    std::size_t read_size = 16 * 1024 * 1024)
+    std::size_t read_size,
+    Args&&... args)
 {
     using namespace detail;
-    File df;
-    File lf;
-    File kf;
+    File df(std::forward<Args>(args)...);
+    File lf(std::forward<Args>(args)...);
```

Comment:
Oh hmm...I see what you mean. The caller can never use `std::move`. How do you suggest this be changed?